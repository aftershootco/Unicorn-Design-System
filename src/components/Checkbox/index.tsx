import clsx from 'clsx'
import React, { useCallback, useState } from 'react'
import styles from './styles.module.scss'

export interface CheckboxProps {
	defaultValue?: boolean
	onChange: (e: boolean) => void
}

const Checkbox: React.FC<CheckboxProps> = ({ defaultValue = false, onChange }) => {
	const [state, setState] = useState(defaultValue)
	const onClick = useCallback(
		() =>
			setState((state) => {
				onChange(!state)
				return !state
			}),
		[setState, onChange]
	)

	return (
		<div className={clsx(styles.checkbox, { [styles.active]: state })} onClick={onClick}>
			<span className={styles.checkmark}></span>
		</div>
	)
}

export default Checkbox
