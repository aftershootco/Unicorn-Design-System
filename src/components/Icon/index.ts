import ClearIcon from './ClearIcon'
import ExpandMore from './ExpandMore'
import ContactSupportIcon from './ContactSupport'
import CheckCircle from './CheckCircle'
import Check from './Check'
import ArrowDownRounded from './ArrowDownRounded'
import ArrrowUpRounded from './ArrrowUpRounded'
import CloseRounded from './CloseRounded'
import ContentCopy from './ContentCopy'
import StarRate from './StarRate'
import ArrowForwardRounded from './ArrowForwardRounded'
import ChevronLeftIcon from './ChevronLeftIcon'
import KeyBoardArrowDown from './KeyBoardArrowDown'
import KeyBoardArrowUp from './KeyBoardArrowUp'
import Menu from './Menu'
import ChevronRightIcon from './ChevronRightIcon'
import KeyBoardLeftIcon from './KeyBoardLeftIcon'
import KeyBoardRightIcon from './KeyBoardRightIcon'

export {
	ClearIcon,
	ExpandMore,
	ContactSupportIcon,
	CheckCircle,
	Check,
	ArrowDownRounded,
	ArrrowUpRounded,
	CloseRounded,
	ContentCopy,
	StarRate,
	ArrowForwardRounded,
	ChevronLeftIcon,
	KeyBoardArrowDown,
	KeyBoardArrowUp,
	Menu,
	ChevronRightIcon,
	KeyBoardLeftIcon,
	KeyBoardRightIcon,
}
