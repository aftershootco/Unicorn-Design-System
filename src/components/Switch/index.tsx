import clsx from "clsx";
import React, { useCallback, useState } from "react";
import styles from "./styles.module.scss";

export interface SwitchProps {
  defaultValue?: boolean;
  onChange: (e: boolean) => void;
}

const Switch: React.FC<SwitchProps> = ({ defaultValue = false, onChange }) => {
  const [state, setState] = useState(defaultValue);
  const onClick = useCallback(() => {
    setState((state) => {
      onChange(!state);
      return !state;
    });
  }, [onChange, setState]);

  return (
    <label
      className={clsx(styles.switch, { [styles.active]: state })}
      onClick={onClick}
    >
      <span className={clsx(styles.round, styles.slider)}></span>
    </label>
  );
};

export default Switch;
