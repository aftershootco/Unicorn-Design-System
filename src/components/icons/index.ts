import ClearIcon from './ClearIcon'
import ExpandMore from './ExpandMore'
import ContactSupportIcon from './ContactSupport'
import CheckCircle from './CheckCircle'
import Check from './Check'
import ArrowDownRounded from './ArrowDownRounded'
import ArrrowUpRounded from './ArrrowUpRounded'
import CloseRounded from './CloseRounded'
import ContentCopy from './ContentCopy'
import StarRate from './StarRate'
import ArrowForwardRounded from './ArrowForwardRounded'
import ChevronLeftIcon from './ChevronLeftIcon'

export {
	ClearIcon,
	ExpandMore,
	ContactSupportIcon,
	CheckCircle,
	Check,
	ArrowDownRounded,
	ArrrowUpRounded,
	CloseRounded,
	ContentCopy,
	StarRate,
	ArrowForwardRounded,
	ChevronLeftIcon,
}
