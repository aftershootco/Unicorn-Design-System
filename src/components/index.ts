import '../index.scss'
import Accordion, { AccordionProps } from './Accordion/Accordion'
import Button, { ButtonProps } from './Button/Button'
import Carousel, { CarouselProps } from './Carousel/Carousel'
import CheckBox, { CheckBoxProps } from './CheckBox/CheckBox'
import Chip, { ChipProps } from './Chip/Chip'
import Divider, { DividerProps } from './Divider/Divider'
import DropDown, { DropdownData, DropDownProps } from './DropDown/DropDown'
import LinearProgressBar, { LinearProgressBarProps } from './LinearProgressBar/LinearProgressBar'
import RadioButtons from './RadioButton/RadioButton'
import Slider from './Slider/Slider'
import Switch, { SwitchProps } from './Switch/Switch'
import Tabs, { TabsProps } from './Tabs/Tabs'
import TextInput, { TextInputProps } from './TextInput/TextInput'
import Toggle, { ToggleProps } from './Toggle/Toggle'
import ToolTip, { ToolTipProps } from './ToolTip/ToolTip'
import HeadlessDropdown, { HeadlessDropdownData, HeadlessDropdownProps } from './HeadlessDropdown/HeadlessDropdown'
import NewDropDown, { NewDropDownProps, NewDropdownData } from './NewDropDown/NewDropDown'

export {
	Accordion,
	AccordionProps,
	Button,
	ButtonProps,
	Carousel,
	CarouselProps,
	Slider,
	DropDown,
	DropdownData,
	DropDownProps,
	ToolTip,
	ToolTipProps,
	Tabs,
	TabsProps,
	CheckBox,
	CheckBoxProps,
	Chip,
	ChipProps,
	Divider,
	DividerProps,
	LinearProgressBar,
	LinearProgressBarProps,
	RadioButtons,
	Switch,
	SwitchProps,
	TextInput,
	TextInputProps,
	Toggle,
	ToggleProps,
	HeadlessDropdown,
	HeadlessDropdownData,
	HeadlessDropdownProps,
	NewDropDown,
	NewDropDownProps,
	NewDropdownData,
}
