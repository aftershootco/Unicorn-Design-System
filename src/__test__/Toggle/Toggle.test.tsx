import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Toggle } from '../../components'

describe(`Testing -> Toggle`, () => {
	it(`to get snapshot`, () => {
		const tree = renderer.create(<Toggle value={true} options={{ first: 'hello', second: 'world' }} onClick={()=>{}}/>).toJSON()
		expect(tree).toMatchSnapshot()
	})

	it(`to be on screen`, () => {
		render(<Toggle value={true} options={{ first: 'hello', second: 'world' }} onClick={() => {}} />)
		expect(screen.getByTestId('test')).toBeInTheDocument()
	})

	it(`to sync the value, options and onClick functionality with true value`, async () => {
		const logSpy = jest.spyOn(console, 'log')
		const { container } = render(
			<Toggle
				value={true}
				options={{ first: 'hello', second: 'world' }}
				onClick={(e) => {
					console.log(e)
				}}
			/>
		)
		const button = container.querySelector('.yearly')
		await userEvent.click(button)
		expect(logSpy).toHaveBeenCalledWith(true)
		expect(button).toHaveTextContent('hello')
	})

	it(`to sync the value, options and onClick functionality with false value`, async () => {
		const logSpy = jest.spyOn(console, 'log')
		const { container } = render(
			<Toggle
				value={false}
				options={{ first: 'hello', second: 'world' }}
				onClick={(e) => {
					console.log(e)
				}}
			/>
		)
		const button = container.querySelector('.monthly')
		await userEvent.click(button)
		expect(logSpy).toHaveBeenCalledWith(false)
		expect(button).toHaveTextContent('world')
	})

	it(`to show the required value in the toggle button`, () => {
		const { container } = render(
			<Toggle
				value={false}
				options={{ first: 'hello', second: 'world' }}
				onClick={(e) => {
					console.log(e)
				}}
			/>
		)
		const button = container.querySelector('.toggle_button')
		expect(button).toHaveClass('toggleButton')
		expect(button).toHaveTextContent('world')
		expect(button).not.toHaveTextContent('hello')
	})
})
