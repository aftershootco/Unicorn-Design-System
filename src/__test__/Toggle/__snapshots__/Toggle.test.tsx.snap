// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing -> Toggle to get snapshot 1`] = `
<div
  className="toggler flex-row"
  data-testid="test"
>
  <button
    className="yearly cursor-pointer text-h4-bold color-white"
    onClick={[Function]}
  >
    hello
  </button>
  <button
    className="monthly cursor-pointer two text-h4-bold color-white"
    onClick={[Function]}
  >
    world
  </button>
  <button
    className="toggle_button text-h4-bold color-white cursor-pointer"
  >
    hello
  </button>
</div>
`;
